pkgs/tools/admin/tigervnc/default.nix

```diff
@@ -76,31 +92,70 @@ stdenv.mkDerivation rec {
         --with-xkb-output=$out/share/X11/xkb/compiled
     make TIGERVNC_SRC=$src TIGERVNC_BUILDDIR=`pwd`/../.. -j$NIX_BUILD_CORES -l$NIX_BUILD_CORES
     popd
+  '' + lib.optionalString stdenv.isDarwin ''
+    make dmg
   '';
 
-  postInstall = ''
+  postInstall = lib.optionalString stdenv.isLinux ''
     pushd unix/xserver/hw/vnc
     make TIGERVNC_SRC=$src TIGERVNC_BUILDDIR=`pwd`/../../../.. install
     popd
     rm -f $out/lib/xorg/protocol.txt
 
     wrapProgram $out/bin/vncserver \
       --prefix PATH : ${lib.makeBinPath (with xorg; [ xterm twm xsetroot xauth ]) }
+  '' + lib.optionalString stdenv.isDarwin ''
+    undmg TigerVNC-$version.dmg
+    mkdir -p $out/Applications
+    mv 'TigerVNC Viewer ${version}.app' $out/Applications/
+    rm $out/bin/vncviewer
+    touch $out/bin/vncviewer
+    echo "#!/bin/bash" >> $out/bin/vncviewer
+    echo "open $out/Applications/TigerVNC\ Viewer\ ${version}.app --args \$@" >> $out/bin/vncviewer
+    chmod +x $out/bin/vncviewer
   '';
 
-  buildInputs = with xorg; [
-    libjpeg_turbo fltk pixman
-    gnutls pam nettle perl
+  buildInputs = [
+    fltk
+    gnutls
+    libjpeg_turbo
+    pixman
+  ] ++ lib.optionals stdenv.isLinux (with xorg; [
+    nettle
+    pam
+    perl
     xorgproto
-    utilmacros libXtst libXext libX11 libXext libICE libXi libSM libXft
-    libxkbfile libXfont2 libpciaccess
+    utilmacros
+    libXtst
+    libXext
+    libX11
+    libXext
+    libICE
+    libXi
+    libSM
+    libXft
+    libxkbfile
+    libXfont2
+    libpciaccess
     libGLU
-  ] ++ xorg.xorgserver.buildInputs;
+  ] ++ xorg.xorgserver.buildInputs
```

Comment:
Why do we need this?

Reply:
I myself am not sure why all the `xorg` logic is structured the way it is -- it seems like a lot of it dates 4 or 6 or more years -- but since the xorg packages were causing build errors on darwin I assumed the simplest approach would to make anything `xorg` related conditional for Linux-only.